src/compiler/binder.ts

```diff
@@ -2555,6 +2555,11 @@ namespace ts {
         // extends clause of a class.
         let transformFlags = subtreeFlags | TransformFlags.AssertES6;
 
+        // propagate ES2017

```

Comment:
Any `TransformFlags` of a subtree that should propagate should already be set in `subtreeFlags`. There should be no need to explicitly propagate.